src/test/kotlin/build/buf/gradle/AbstractGenerateTest.kt

```diff
@@ -16,10 +16,31 @@ package build.buf.gradle
 
 import com.google.common.truth.Truth.assertThat
 import org.gradle.language.base.plugins.LifecycleBasePlugin.BUILD_TASK_NAME
+import org.junit.jupiter.api.BeforeEach
 import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInfo
+import java.io.File
+import java.nio.file.Files
 import java.nio.file.Paths
+import kotlin.io.path.isRegularFile
+import kotlin.io.path.name
 
 abstract class AbstractGenerateTest : AbstractBufIntegrationTest() {
+    @BeforeEach
+    override fun setup(testInfo: TestInfo) {
+        super.setup(testInfo)
+        val protocPath = System.getProperty("protoc.path") ?: throw RuntimeException("protoc.path not set")
+        val protocFile = File(protocPath)
+        if (!protocFile.canExecute()) {
+            protocFile.setExecutable(true)
+        }
+        Files.walk(super.projectDir.toPath()).forEach { path ->
```

Comment:
Consider [`super.projectDir.walkTopDown`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/java.io.-file/walk-top-down.html)?

Reply:
Done in https://github.com/bufbuild/buf-gradle-plugin/pull/230/commits/f6dc91abdf4179571522eaadebcb6b9a95832843.